data_diff/sqeleton/databases/snowflake.py

```diff
@@ -142,8 +142,27 @@ def _query(self, sql_code: Union[str, ThreadLocalInterpreter]):
         return self._query_conn(self._conn, sql_code)
 
     def select_table_schema(self, path: DbPath) -> str:
-        schema, name = self._normalize_table_path(path)
-        return super().select_table_schema((schema, name))
+        """Provide SQL for selecting the table schema as (name, type, date_prec, num_prec)"""
+        database, schema, name = self._normalize_table_path(path)
+
+        return (
+            "SELECT column_name, data_type, datetime_precision, numeric_precision, numeric_scale "
+            f"FROM {database + '.' if database else ''}information_schema.columns "
+            f"WHERE table_name = '{name}' AND table_schema = '{schema}'"
+        )
+
+    def _normalize_table_path(self, path: DbPath) -> DbPath:
+        if len(path) == 1:
+            if self.default_schema:
```

Comment:
What if default_schema is None?

Reply:
this is the logic from base.py https://github.com/datafold/data-diff/blob/6f5417a179120dbf7d4eb96055eaf8670e0e5979/data_diff/sqeleton/databases/base.py#L415-L419

When schema is None in the constructor, an exception is thrown here https://github.com/datafold/data-diff/blob/6f5417a179120dbf7d4eb96055eaf8670e0e5979/data_diff/sqeleton/databases/snowflake.py#L114
```
File "/Users/dan/repos/data-diff/data_diff/sqeleton/databases/snowflake.py", line 114, in __init__
    assert '"' not in schema, "Schema name should not contain quotes!"
TypeError: argument of type 'NoneType' is not iterable
```

That is an inaccurate exception, and is an example of where unit tests would be useful vs. the current tests, but it's also not related to the change.

Reply:
You make a good point, but the current implementation in `base.py` is actually a bug, which our drivers just happen to protect against.

You don't have to fix it in this PR, but you also shouldn't copy that bug into a new function.

If you expect `_normalize_table_path()` to return a 3-tuple, then it should always do so.

Reply:
FWIW, I'm now fixing the base implementation: https://github.com/datafold/data-diff/pull/333

Reply:
updated